python/ray/serve/api.py

```diff
@@ -661,6 +655,14 @@ def __init__(self,
         if init_args is None:
             init_args = ()
 
+        # TODO(architkulkarni): Enforce that autoscaling_config and
+        # user-provided num_replicas should be mutually exclusive.
+        if version is None and config.autoscaling_config is not None:
+            # TODO(architkulkarni): Remove this restriction.
+            raise ValueError(
+                "Currently autoscaling is only supported for "
+                "versioned deployments. Try @serve.deployment(version=...).")
```

Comment:
Hmm I think we should be able to get around this pretty easily by adding a new internal API to not auto-generate a version. Fine with doing it in a follow-up PR but this is a basic acceptance criteria.